server/controllers/personal/personalController.ts

```diff
@@ -150,10 +166,35 @@ export default class PersonalController {
           const feet = parseInt(feetString, 10)
           const inches = parseInt(inchesString, 10)
 
-          if (Number.isNaN(feet) || feet <= 0 || Number.isNaN(inches) || inches <= 0) {
+          const validations = [
+            {
+              isInvalid: () =>
+                ((feetString === '' || feetString === undefined) &&
+                  (inchesString === '' || inchesString === undefined)) ||
+                feet < 1 ||
+                feet > 9 ||
+                (feet === 9 && inches > 0),
+              message: 'Height must be between 1 feet and 9 feet',
+            },
+            {
+              isInvalid: () =>
+                feetString === '' ||
+                feetString === undefined ||
+                Number.isNaN(feet) ||
+                Number.isNaN(inches) ||
+                (feet >= 1 && feet <= 9 && inches < 0),
+              message: 'Feet must be between 1 and 9. Inches must be between 0 and 11',
+            },
+          ]
+
+          const errors = validations.find(({ isInvalid }) => {
+            return isInvalid()
+          })
```

Comment:
I may be being dim... not sure I follow this. Both validations hardcoded to contain an isInvalid function, so it will always find the first one. As far as I can see errors === validations[0].isInvalid()

Reply:
It calls the function so it's the result, if both are false then `find()` errors will be `undefined`

Reply:
I have realised that there's a nicer way to do this anyway so i'm about to change this anyway aha